### ISSUE TITLE
Initial cluster creation fails to init node on rootless Docker.

### ISSUE THREAD

**Issue by SanjayVas** [kubernetes-sigs/kind#2541]
**What happened**: Cluster creation failed with `ERROR: failed to create cluster: failed to init node with kubeadm: command "docker exec --privileged kind-control-plane kubeadm init --skip-phases=preflight --config=/kind/kubeadm.conf --skip-token-print --v=6" failed with error: exit status 1`. Full error output below.

**What you expected to happen**: Cluster creation to succeed.

**How to reproduce it (as minimally and precisely as possible)**:
1. Follow configuration instructions from https://kind.sigs.k8s.io/docs/user/rootless/ to set up Kind to run with rootless Docker.
2. Run `kind create cluster`.

**Anything else we need to know?**:

- Output of `kind create cluster`: [kind.log](https://github.com/kubernetes-sigs/kind/files/7542354/kind.log)
- Output of `docker logs kind-control-plane`: [kind-control-plane.log](https://github.com/kubernetes-sigs/kind/files/7542575/kind-control-plane.log)
- 
**Environment**:

- kind version: (use `kind version`): `kind v0.11.1 go1.16.4 linux/amd64`
- Kubernetes version: (use `kubectl version`): `Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"clean", BuildDate:"2021-10-27T18:41:28Z", GoVersion:"go1.16.9", Compiler:"gc", Platform:"linux/amd64"}`
- Docker version: (use `docker info`): (edited to remove potentially sensitive information)
```
Client:
 Context:    default
 Debug Mode: false

Server:
 Containers: 6
  Running: 0
  Paused: 0
  Stopped: 6
 Images: 2
 Server Version: 20.10.10
 Storage Driver: fuse-overlayfs
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runtime.v1.linux runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 5b46e404f6b9f661a205e28d59c982d3634148f8
 runc version: v1.0.2-0-g52b36a2d
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  rootless
  cgroupns
 Kernel Version: 5.10
 Operating System: Debian GNU/Linux
 OSType: linux
 Architecture: x86_64
 CPUs: 12
 Total Memory: 62.41GiB
 ID: 2PP4:7UAI:C6LW:577S:XZVM:UPYJ:FITI:TMJ3:WXI3:6RBV:55UD:2OI7
 Debug Mode: false
 Username: sanjayvas
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine

WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```
- OS (e.g. from `/etc/os-release`): Debian 12 Bookworm

**Comment by AkihiroSuda** [comment]
`docker exec kind-control-plane journalctl` may print more detailed logs.

Also, could you try the main branch?

**Comment by SanjayVas** [comment]
Using kind built from main resolved the issue for me.

**Comment by BenTheElder** [comment]
This is probably https://github.com/kubernetes-sigs/kind/issues/2275, currently the next release is blocked on other regressions that've occurred since being fixed.

**Comment by SanjayVas** [comment]
I'm now getting an error when trying to load an image archive. Let me know if you'd prefer me to file a separate issue for this.

```
$ kind load image-archive src/main/kotlin/org/wfanet/measurement/duchy/deploy/common/server/async_computation_control_server_image.tar -v 1
ERROR: failed to load image: command "docker exec --privileged -i kind-control-plane ctr --namespace=k8s.io images import -" failed with error: exit status 1
Command Output: unpacking docker.io/bazel/src/main/kotlin/org/wfanet/measurement/duchy/deploy/common/server:async_computation_control_server_image (sha256:72adfdb270f0a3a69b425fe725c3028ec72b1d6227046ce31f2ff3baf0f2e443)...time="2021-11-16T19:53:02Z" level=info msg="apply failure, attempting cleanup" error="failed to extract layer sha256:e3d24823466584f8fcc6fe0d7d5bb09d2360b86ef9b655e21e8002ee04dd386b: failed to mount /var/lib/containerd/tmpmounts/containerd-mount736258380: operation not permitted: unknown" key="extract-834705651-6zVb sha256:5c30f10f05a6636f4b4a9fe393b48215dd35657cf972784e74027e488f9cbd0f"
ctr: failed to extract layer sha256:e3d24823466584f8fcc6fe0d7d5bb09d2360b86ef9b655e21e8002ee04dd386b: failed to mount /var/lib/containerd/tmpmounts/containerd-mount736258380: operation not permitted: unknown
Stack Trace:
sigs.k8s.io/kind/pkg/errors.WithStack
        /usr/local/google/home/sanjayvas/go/pkg/mod/sigs.k8s.io/kind@v0.11.2-0.20211115230152-60143cb4e28c/pkg/errors/errors.go:59
sigs.k8s.io/kind/pkg/exec.(*LocalCmd).Run
        /usr/local/google/home/sanjayvas/go/pkg/mod/sigs.k8s.io/kind@v0.11.2-0.20211115230152-60143cb4e28c/pkg/exec/local.go:124
sigs.k8s.io/kind/pkg/cluster/internal/providers/docker.(*nodeCmd).Run
        /usr/local/google/home/sanjayvas/go/pkg/mod/sigs.k8s.io/kind@v0.11.2-0.20211115230152-60143cb4e28c/pkg/cluster/internal/providers/docker/node.go:146
sigs.k8s.io/kind/pkg/cluster/nodeutils.LoadImageArchive
        /usr/local/google/home/sanjayvas/go/pkg/mod/sigs.k8s.io/kind@v0.11.2-0.20211115230152-60143cb4e28c/pkg/cluster/nodeutils/util.go:80
sigs.k8s.io/kind/pkg/cmd/kind/load/image-archive.loadImage
        /usr/local/google/home/sanjayvas/go/pkg/mod/sigs.k8s.io/kind@v0.11.2-0.20211115230152-60143cb4e28c/pkg/cmd/kind/load/image-archive/image-archive.go:136
sigs.k8s.io/kind/pkg/cmd/kind/load/image-archive.runE.func1
        /usr/local/google/home/sanjayvas/go/pkg/mod/sigs.k8s.io/kind@v0.11.2-0.20211115230152-60143cb4e28c/pkg/cmd/kind/load/image-archive/image-archive.go:123
sigs.k8s.io/kind/pkg/errors.UntilErrorConcurrent.func1
        /usr/local/google/home/sanjayvas/go/pkg/mod/sigs.k8s.io/kind@v0.11.2-0.20211115230152-60143cb4e28c/pkg/errors/concurrent.go:30
runtime.goexit
        /usr/lib/google-golang/src/runtime/asm_amd64.s:158
```

I get the same error if I load the archive into Docker first and use `kind load docker-image`, though I have to remove it first using `docker exec -it kind-control-plane crictl rmi` as it thinks it's "found to be already present on all nodes" despite the failure. I can run a container using that image just fine in Docker.

**Comment by BenTheElder** [comment]
> as it thinks it's "found to be already present on all nodes" despite the failure.

this is a problem with containerd, it will import the image metadata and report that the image exists by that hash even if the layers are missing. I don't think we can do much about this, maybe a `--force` option, but that wouldn't help here since unpacking will just fail again anyhow.

the layer failing to unpack is problematic, can you please file a different issue for this with as much detail as possible? (filing a different issue will help future users looking for the same issue).

**Comment by BenTheElder** [comment]
the containerd logs should be helpful here.

**Comment by BenTheElder** [comment]
Closing since the original issue is fixed @ HEAD, but it seems we still have some unpacking issue in https://github.com/kubernetes-sigs/kind/issues/2543 (thanks!)
